### PR TITLE
feat(proto,cli): add Galaxy server config as scan metadata (ADR-045 PR2)

### PR DESCRIPTION
**Pull request by djdanielsson**
## Summary

- Add `GalaxyServerDef` proto message and `galaxy_servers` fields to `ScanOptions`/`FixOptions`, enabling Galaxy/Automation Hub credentials to flow as scan-scoped metadata through gRPC (ADR-045 PR2)
- CLI automatically discovers Galaxy servers from the user's `ansible.cfg` — zero configuration required for existing Ansible users
- Primary writes a session-scoped temp `ansible.cfg` from the proto fields, ready for proxy consumption

### Proto
- `common.proto`: new `GalaxyServerDef` message (`name`, `url`, `token`, `auth_url`)
- `primary.proto`: `repeated GalaxyServerDef galaxy_servers` on `ScanOptions` (field 5) and `FixOptions` (field 9)
- Regenerated gRPC stubs with grpcio-tools 1.78.0 (matching project pin); updated `.pyi` type stubs

### CLI
- New `_galaxy_config.py` module: `resolve_ansible_cfg()` follows Ansible's own precedence (`ANSIBLE_CONFIG` env → project `ansible.cfg` → `~/.ansible.cfg`), `parse_galaxy_servers()` reads `[galaxy] server_list` + `[galaxy_server.<name>]` sections into `GalaxyServerDef` protos
- `check.py` / `remediate.py`: call `discover_galaxy_servers()` and populate `galaxy_servers` on scan/fix options
- `chunked_fs.py`: `build_scan_bundle()` and `yield_scan_chunks()` accept and propagate `galaxy_servers`

### Primary server
- `_write_session_galaxy_cfg()`: converts proto `GalaxyServerDef` → `GalaxyServerConfig` → writes temp `ansible.cfg` via existing `write_temp_ansible_cfg()`
  - Strips whitespace from URLs before filtering
  - Deduplicates server names with index suffix to prevent ValueError
  - Wrapped in try/except so Galaxy config failures don't abort sessions
- `_session_process()`: extracts `galaxy_servers` from `fix_opts` or `scan_opts`, writes session-scoped config
- `SessionState`: new `galaxy_cfg_path` field, cleaned up on session end
- `_scan_pipeline()`: `galaxy_cfg_path` parameter threaded through for future proxy integration

### Docs
- `DATA_FLOW.md`: updated ScanOptions field list to include `galaxy_servers`

### Tests (25 new)
- `test_galaxy_config.py` (15): config resolution, parsing edge cases, proto type checks
- `test_galaxy_proto_wiring.py` (10): proto roundtrip, ScanOptions/FixOptions population, session config writer, whitespace URL filtering, name deduplication

## Test plan

- [x] `prek run --all-files` passes (all 6 hooks)
- [x] 25 new tests pass
- [x] 106 related existing tests pass (collection_downloader, session, project_root, galaxy_proxy_server)
- [x] No docs regressions

### CODE REVIEW

File: src/apme_engine/daemon/primary_server.py [hunk 1]
```diff
@@ -99,6 +100,46 @@ class _ValidatorResult:
...
+    configs = [
+        GalaxyServerConfig(
+            name=s.name or f"server_{i}",
+            url=s.url,
+            token=s.token or None,
+            auth_url=s.auth_url or None,
+        )
+        for i, s in enumerate(galaxy_servers)
+        if s.url
+    ]
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply src/apme_engine/daemon/primary_server.py]: Now stripping URLs with `(s.url or "").strip()` before filtering, so whitespace-only URLs are properly skipped. Added a test for this case. Fixed in 65c76c3.

File: src/apme_engine/daemon/primary_server.py [hunk 2]
```diff
@@ -99,6 +100,46 @@ class _ValidatorResult:
...
+    if not configs:
+        return None
+
+    cfg_dir = Path(tempfile.mkdtemp(prefix="apme-galaxy-session-"))
+    return write_temp_ansible_cfg(configs, cfg_dir)
```
[the remark this answers is not in the document]
Comment (djdanielsson) [reply src/apme_engine/daemon/primary_server.py]: Wrapped in `try/except Exception` with `logger.exception()` so Galaxy config failures are logged but don't abort the session. Fixed in 65c76c3.

**Review by cidrblock — APPROVED** [review]
Clean, well-structured PR. Galaxy server credentials flow correctly from ansible.cfg through the gRPC contract to Primary's session-scoped temp config. Good scope control — gets the data where it needs to be without prematurely wiring the proxy.

Highlights:
- Proto design is clean; field numbers are conflict-free
- CLI config discovery correctly follows Ansible's own resolution order with `interpolation=None`
- Defensive coding in Primary: URL whitespace stripping, name deduplication, try/except wrapping, temp dir cleanup
- 25 tests cover all the boundary cases
- All Copilot feedback addressed (grpcio-tools version, dedup, whitespace, error handling)

Minor notes (non-blocking):
- `GalaxyServerDef.token` is plaintext — fine for local pod transport, worth flagging if this ever travels over the network
- `galaxy_cfg_path` is threaded through `_scan_pipeline` but not yet consumed (intentional, PR2 scope)

LGTM — approve and merge.
